langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/internal/AbstractSharedBedrockChatModel.java

```diff
@@ -0,0 +1,111 @@
+package dev.langchain4j.model.bedrock.internal;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageType;
+import dev.langchain4j.internal.Json;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeClient;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.joining;
+
+@Getter
+@SuperBuilder
+public class AbstractSharedBedrockChatModel {
```

Comment:
[Sugesstion] 

Instead of the shared model and then extending the other classes with complex inheritance. I think we could use composition (at least in the scope of our changes).

I see such an implementation with a util class here: https://github.com/beekpr/langchain4j/blob/main/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/OllamaMessagesUtils.java

Reply:
@anandbhaskaran that was my first thought, but then I didn't see this pattern in other libraries, I'll check maybe how other integrations are written.